sw/host/opentitanlib/src/image/manifest_ext.rs

```diff
@@ -86,11 +93,39 @@ impl ManifestExtSpec {
     pub fn source_path(&self) -> Option<&Path> {
         self.relative_path.as_deref()
     }
+
+    pub fn check_for_duplicates(&self, id: ManifestExtId) -> Result<()> {
```

Comment:
nit: the name doesn't match what the function does. Should we call this `exists` or `extension_present` or something like that?